lib/ex_unit/extra.ex

```diff
@@ -25,4 +25,38 @@ defmodule ExUnit.Extra do
     end
   end
 
+
+  @doc """
+  Compares two MapSets and asserts that they are equal.
+
+  Both arguments are passed through MapSet.new/1 before being compared.
+
+    iex> import ExUnit.Extra
+    ...> assert_set_equality [:a, :b], [:b, :a]
+    true
+
+  """
+  defmacro assert_set_equality(set_a, set_b) do
+    quote do
+      assert MapSet.new(unquote(set_a)) == MapSet.new(unquote(set_b))
+    end
+  end
+
+
+  @doc """
+  Compares two MapSets and asserts that they are NOT equal.
+
+  Both arguments are passed through MapSet.new/1 before being compared.
+
+    iex> import ExUnit.Extra
+    ...> refute_set_equality [:a, :b], [:a]
+    false
```

Comment:
Are these examples correct? Doesn't `assert` throw an exception if it fails?

Reply:
It does, hence the use of refute here – it will throw if the passed sets are equal. It is strange that it returns false, but it's literally just a `not assert/2`, which returns true.

Thanks for making me check though - i'd completely forgotten to add the `doctest ExUnit.Extra`, so this wasn't running at all!